src/compiler/checker.ts

```diff
@@ -8765,7 +8765,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                                     /*modifiers*/ undefined,
                                     getEffectiveDotDotDotForParameter(p),
                                     setTextRange(context, factory.createIdentifier(getNameForJSDocFunctionParameter(p, i)), p),
-                                    p.questionToken,
+                                    p.questionToken ? factory.createToken(SyntaxKind.QuestionToken) : undefined,

```

Comment:
Does `factory.cloneNode` not work on tokens?

Reply:
It does but this short circuits some of its logic, we know exactly what we want here. Unless those should have `.original` attached to them and such - if that's the case then it would be easier to use `cloneNode`. I went this route since some other code path here was already doing it like this, IIRC

Reply:
Being consistent is fine with other code around it (I didn't look around)